store/src/java/com/zimbra/soap/SoapServlet.java

```diff
@@ -204,11 +202,27 @@ public static void addService(String servletName, DocumentService service) {
         }
     }
 
+    public static void removeService(String servletName, RemovableDocumentService service) {
+        synchronized (sExtraServices) {
+            ZimbraServlet servlet = ZimbraServlet.getServlet(servletName);
+            if (servlet != null) {
+                ((SoapServlet) servlet).removeService(service);
+            } else {
+                sLog.info("No servlet found with name: %s", servletName);
+            }
+        }
+    }
+
     private void addService(DocumentService service) {
         ZimbraLog.soap.info("Adding service %s to %s", service.getClass().getSimpleName(), getServletName());
         service.registerHandlers(mEngine.getDocumentDispatcher());
     }
 
+    private void removeService(RemovableDocumentService service) {
+        ZimbraLog.soap.info("Removing service %s to %s", service.getClass().getSimpleName(), getServletName());
```

Comment:
log message isn't clear, what details are we capturing here?

Reply:
Only the service class name and service name